src/services/metrics.py

```diff
@@ -121,3 +122,17 @@ async def monitor_http_requests(self, req: Request, call_next: Any) -> Any:
 
         # continue request handling.
         return response
+
+
+def get_custom_metrics() -> Generator[CustomMetrics, None, None]:
```

Comment:
What is the purpose of `Generator` here? Why it can't be a simple `return`?

Reply:
So glad that you ask. It is no longer needed. I restructured the code but overlooked that this would make the generators obsolete. The original intention to close connections in a finally block. But since that is not done in any other place where we open connections, I adapted to that.